src/plugins/vis_builder/public/types.ts

```diff
@@ -37,6 +37,7 @@ export interface VisBuilderPluginStartDependencies {
 }
 
 export interface VisBuilderServices extends CoreStart {
+  appName: string;
```

Comment:
The changes to set `appName` in VisBuilderServices seems fine, but it's unclear why it's necessary for this PR. I didn't see it used in `secondary_panel.tsx`?

Reply:
Ah this is my bad. For the `filters` aggregation to work, it relies on the app name and storage option in services. Since app name was already being referenced in other places, I decided to add it to the root level service and simply reference that everywhere else too. 

The filters agg uses storage (here it is local storage) with the app name as key to save and retrieve the history of all queries.